File: mobile/lib/services/tag.dart
import '../common/constants.dart';
import '../common/models.dart';
import 'api_client.dart';

const String _path = '/tag';

class TagService {
  final ApiClient _apiClient = ApiClient();

  Future<Map<String, dynamic>> create(String name,) async {
    final body = {
      kName: name,
    };
    return await _apiClient.post(_path, body: body);
  }

  Future<List<MetricDetails>> list({Map<String, String>? queryParams}) async {
    final response = await _apiClient.get(_path, queryParams: queryParams);
    return response.map((metric) => MetricDetails.fromJson(metric)).toList();
  }

}

File: e2e/clients/tag.py
from typing import Dict, Any, List

import api
from e2e.common import base_url, build_query_string
from shared import constants

tag_path = base_url + '/tag'


def create(name: str, jwt: str) -> int:
    return api.create(
        tag_path, {
            constants.name: name,
        }, jwt)


def get(jwt: str, query_params: Dict[str, str] = {}, ) -> Dict[str, Any]:
    return api.get(tag_path + f'?{build_query_string(query_params)}', jwt)

